packages/collaboration/src/sharedlink.ts

```diff
@@ -45,22 +45,458 @@ export async function showSharedLinkDialog({
+      'beforeend',
+      this._trans.__(
+        'Warning: Anyone you share this server with will have access to all your files, not just the currently open file.'
+      )
+    );
+    dialogBody.insertAdjacentHTML('beforeend', '<br>');
+    dialogBody.insertAdjacentText(
+      'beforeend',
+      this._trans.__(
+        'If your Hub administrator allows it, you can create another server dedicated to sharing a specific project.'
+      )
+    );
+  }
+
+  private updateSearchResults(): void {
+    if (!this._searchResults) {
+      return;
+    }
+    this._searchResults.innerHTML = '';
+    const query = this._searchInput?.value || '';
+
+    const filteredUsers = this._users.filter(
+      user =>
+        user.name.toLowerCase().includes(query.toLowerCase()) || query === ''
+    );
+
+    filteredUsers.forEach(user => {
+      const userElement = document.createElement('div');
+      userElement.classList.add('user-item');
+      userElement.textContent = user.name;
+      userElement.addEventListener('click', async () => {
+        await this.createShare(user, user.type);
+        await this.loadShares();
+        await this.updateSharesList();
+        // Removing the new user from the search results
+        const sharedUserNames = new Set(this._shares.map(share => share.name));
+        this._users = this._users.filter(
+          (user: any) =>
+            user.name !== this._serverOwner && !sharedUserNames.has(user.name)
+        );
+        this.updateSearchResults();
+      });
+      this._searchResults?.appendChild(userElement);
+    });
+  }
+
+  private async updateSharesList(): Promise<void> {
+    if (!this._sharesContainer) {
+      return;
+    }
+    this._sharesContainer.innerHTML = '';
+
+    const table = document.createElement('table');
+    table.classList.add('shares-table');
+
+    const headerRow = document.createElement('tr');
+    headerRow.innerHTML = `
+      <th>${this._trans.__('Shared with')}</th>
+      <th>${this._trans.__('Shared since')}</th>
+      <th>${this._trans.__('Actions')}</th>
+    `;
+    table.appendChild(headerRow);
+
+    if (this._shares.length === 0) {
+      const row = document.createElement('tr');
+      const cell = document.createElement('td');
+      cell.colSpan = 3;
+      cell.textContent = this._trans.__(
+        'Your server is not shared to anybody yet. You can search for users and groups above.'
+      );
+      row.appendChild(cell);
+      table.appendChild(row);
+    } else {
+      this._shares.forEach(share => {
+        const row = document.createElement('tr');
+        const formattedDate = new Date(share.createdAt).toLocaleString([], {
+          hour: '2-digit',
+          minute: '2-digit',
+          year: 'numeric',
+          month: '2-digit',
+          day: '2-digit'
+        });
+
+        row.innerHTML = `
+        <td>${
+          share.type === 'group'
+            ? this._trans.__('Group ') + share.name
+            : share.name
```

Comment:
I worry that this might lead to script injection is `share.name` is comporomised. As a rule we should avoid `innerHTML` as much as possible. Can you instead create this cells with `document.createElement`?